pandas/core/util/numba_.py

```diff
@@ -11,6 +11,14 @@
 
 from pandas.util.version import Version
 
+try:
+    # Reduce likelihood of segfaulting during parallelism
+    # https://github.com/numba/numba/issues/6839#issuecomment-868352174
+    numba = import_optional_dependency("numba")
+    numba.config.THREADING_LAYER = "threadsafe"
```

Comment:
I don't think this will have any effect if something has already triggered a `parallel=True` compilation elsewhere (e.g. user code).